src/System.Windows.Forms/src/System/Windows/Forms/Layout/DockAndAnchorLayout.cs

```diff
@@ -748,7 +748,10 @@ public static void SetAnchor(IArrangedElement container, IArrangedElement elemen
 
                 CommonProperties.xSetAnchor(element, value);
 
-                if (CommonProperties.GetNeedsAnchorLayout(element))
+                // Updating AnchoriInfo is only needed when control is ready for layout. InitLayoutCore() does call UpdateAnchorInfo().
+                // At the least, we are checking if control is parented before updating AnchorInfo. This helps avoid calculating
+                // AnchorInfo with default initial values of the Control. They are always overriden when layout happen.
```

Comment:
Few comments on the comment:
* I'm struggling to draw the connection between the first and the second sentences: "_Updating AnchoriInfo is only needed when control is ready for layout. InitLayoutCore() does call UpdateAnchorInfo()._". Do you think you could help me understand the connection?
* "_At the least,_" doesn't look like it fits in the context too, as it kind of implies "one of the above options", but in the code we're doing "and" operation. Perhaps "_In addition_" would work better here?
* "_They are always overriden when layout happen._" - to me "override" sounds like we're overriding a method. Were you meant to say something that has "updated" or "recalculated" meaning?

```suggestion
                // AnchorInfo only needs to be updated when the control is ready for layout. InitLayoutCore() call UpdateAnchorInfo().
                // In addition we need to check whether the control is parented before updating AnchorInfo. This helps to avoid calculating
                // AnchorInfo with the control's default initial values, as those are always overriden when layout is performed.
```